pkg/task/task_manager.go

```diff
@@ -69,28 +72,36 @@ func (m *TaskManager) Submit(f func() error) string {
 	return TaskId
 }
 
-// TODO currently result is only available for once, need store until clean
 func (m *TaskManager) GetTaskResult(taskId string) (*TaskResult, error) {
 	retCh, exists := m.ResultMap[taskId]
 	if !exists {
 		// m.Logger.Info("Query a task id that's not exists", "task id", taskId)
 		return nil, errors.Errorf("Task %s not exists", taskId)
 	}
-	select {
-	case result := <-retCh:
-		return result, nil
-	default:
-		return nil, nil
+	if m.TaskResultCache[taskId] == nil {
+		select {
+		case result := <-retCh:
+			m.TaskResultCache[taskId] = result
+			return result, nil
+		default:
+			return nil, nil
+		}
+	} else {
+		return m.TaskResultCache[taskId], nil
 	}
 }
 
 func (m *TaskManager) CleanTaskResult(taskId string) error {
 	retCh, exists := m.ResultMap[taskId]
 	if !exists {
 		return errors.Errorf("Task %s not exists", taskId)
-		// m.Logger.Error(err, "Task not exists", "task id", taskId)
+	}
+	_, exists = m.TaskResultCache[taskId]
```

Comment:
there's no task result in TaskResultCache if GetTaskResult has not been called

Reply:
This check is not really necessary, can serve as a robust check.